### ISSUE TITLE
KeePassXC 2.7.1: File change date of .kdbx file falls back to previous file date

### ISSUE THREAD

**Issue by Reefkeeper** [keepassxreboot/keepassxc#8034]
## Overview
When writing changes to the KDBX file the file change date sometimes is not the current date/time but a previous file date/time is restored. I noted this in version 2.7.1. Had Keepass (not XC) before for many years without this problem.

## Steps to Reproduce

1. Open an existing file with filedate A
2. Make one change (i. e. add comment) to any item. Watch file date: it changes to current date/time
3. Make another change, press OK button and watch file directory and file dates of KDBX file and temporary file.watch file date again: it returns to original date/time
4. Once this state is reached, the only help to get out of this problem is to have another KeePass-App (KeePass2Android in my case) make changes to the KDBX-file. After this change, KeePassXC will change the file date once correctly but will fall back to the issue with further changes.

## Expected Behavior
KDBX file has newer, current date/time, at least not the original filedate A.

## Actual Behavior
Every now and then, the file date is not the current file date/time but the file date/time of the original file, i. e. from the previous file.
This doesn't happen every time, but when it starts to happen, additional changes and saves to the database do not correct this error.
The file date then remains with the original filedate A. 

It also happens that at first a correct, actual file date/time is applied, everythings seems ok. But after 1 or 2 more changes in the KDBX file the old file date/time is restored and it remains old.
Exiting KeePassXC and restarting and changing items and saving them again does not help. The file date remains old.

## Context
I had Keepass (not XC) before for many years without this problem.

KeePassXC - Version 2.7.1
Revision: 5916a8f
Distribution: AppImage

Qt 5.15.2
Diagnosemodus ist deaktiviert.

Betriebssystem: Linux Mint 20.3
CPU-Architektur: x86_64
Kernel: linux 5.4.0-80-generic

Aktivierte Erweiterungen:
- Auto-Type
- Browser-Integration
- SSH-Agent
- KeeShare
- YubiKey
- Secret-Service-Integration

Kryptographische Bibliotheken:
- Botan 2.12.1

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linxu Mint 20.3
Desktop Env: Cinnamon

**Comment by droidmonkey** [comment]
The intended behavior is to retain the file **creation time** and update the **modified time**. It is hard to follow your sequencing above and I am not clear as to which dates you are referring to. The code to retain the file creation time was introduced in 2.6.5 from #6576

**Comment by Reefkeeper** [comment]
Hello droidmonkey, thank you for reply! 
In fact I am referring to the modified time. I use this time for two-way syncing the database from my Fritz-NAS USB stick to 2 Android phones via the app FolderSyncPro. Hence the modified time is very important. Due to the potential bug I've already lost 1 password which I had changed and which was overwritten by an older version.
This syncing methode has worked for more than 2 years with KeePass (w/o XC) without problems. I use KeePassXC for 2 weeks now.

I have now tested quite a while and found the following: I couldn't reproduce this error on a normal home directory (fast hard drive) but had the error very often when writing and reading from Fritz NAS USB stick. It doesn't matter wether I use the USB stick or save the file on the Fritzbox NAS itself (internal memory). I can see - due to the slower Fritz memory that when the file is rewritten, the following happens. Let's assume the file name is  "big_github.kdbx". 
I press the OK button to save changes:
1. An additional file with the naming convention  "big_gihub.kdbx.<RandomName>" with the current time is created.
2. This new file is immediately deleted again.
3. Then either of these happen:
    a)  _Normal case:_  The file "big_github.kdbx" gets a new time. OK
    b)  _Faulty case_: The file "big_github.kdbx" gets a older time **- it keeps its old time or gets an even older filedate!**

I have added 2 short videos with the "normal" and the "faulty" case. 
Each time I make a change to the database and hit the OK button.
Watching the videos I suspect that a timing problem may lead to the wrong date.

https://user-images.githubusercontent.com/18089056/167881053-24409412-c649-4a1d-ac4e-1f94b30c59a0.mp4


https://user-images.githubusercontent.com/18089056/167881107-38f8e36c-4bdd-44d3-a094-54a09b4673bf.mp4

**Comment by Reefkeeper** [comment]
I found a contradiction between modified file date/time and the file date/time of an entry within this database file, s. photo attached:
The file date/time is older than the entry date:
![Difference file date and date of note change](https://user-images.githubusercontent.com/18089056/167897789-07c3af5e-1b50-4208-84d9-c0c4728a21be.png)
.

**Comment by droidmonkey** [comment]
Recommend enabling direct write saves in the application settings.

**Comment by Reefkeeper** [comment]
Changed it to direct write saves (labelled "dangerous") **-> seems to work!!!!**!

Do you happen to know wether this method is standard/default in KeePass?
Anyway, now I can sleep much better!!!!
Thank you very much!!!

**Comment by Reefkeeper** [comment]
With the above save methode, would you recommend to enable the automatic backup file creation or does this interfere with the direct save method? I can see that with this option enabled there are quite many file operations.
Thank you!

**Comment by Reefkeeper** [comment]
Sorry, to bother again, I can answer my last question to myself: the backup option is not recommendable, it leads to the same problems as above.
Thank you again!

**Comment by droidmonkey** [comment]
Ok great, since you already have some sort of backup method that should save you in case of a fault.
